homeassistant/components/esphome/assist_satellite.py

```diff
@@ -253,6 +262,21 @@ def on_pipeline_event(self, event: PipelineEvent) -> None:
 
         self.cli.send_voice_assistant_event(event_type, data_to_send)
 
+    async def async_announce(self, message: str, media_id: str) -> None:
+        """Announce media on the satellite.
+
+        Should block until the announcement is done playing.
+        """
+        self._announcement_finished.clear()
```

Comment:
Clear feels weird. I would do this:
```suggestion
        if self._announcement_finished:
            self._announcement_finished.set()

        event = self._announcement_finished = asyncio.Event()
```

Reply:
Or, since we know the other one will be aborted, should there be some raising happening?